networkx/algorithms/community/bipartitions.py

```diff
@@ -0,0 +1,351 @@
+    But they claim that can make a difference between a good and excellent method.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Graph must be undirected.
+
+    partition : tuple
+        Pair of iterables containing an initial partition. If not
+        specified, a random balanced partition is used.
+
+    max_iter : int
+        Maximum number of times to attempt swaps to find an
+        improvement before giving up.
+
+    weight : string or function (default: "weight")
+        If this is a string, then edge weights will be accessed via the
+        edge attribute with this key (that is, the weight of the edge
+        joining `u` to `v` will be ``G.edges[u, v][weight]``). If no
+        such edge attribute exists, the weight of the edge is assumed to
+        be one.
+
+        If this is a function, the weight of an edge is the value
+        returned by the function. The function must accept exactly three
+        positional arguments: the two endpoints of an edge and the
+        dictionary of edge attributes for that edge. The function must
+        return a number or None to indicate a hidden edge.
+
+    seed : integer, random_state, or None (default)
+        Indicator of random number generation state.
+        See :ref:`Randomness<randomness>`.
+        Only used if partition is None
+
+    Returns
+    -------
+    partition : tuple
+        A pair of sets of nodes representing the bipartition.
+
+    Raises
+    ------
+    NetworkXError
+        If partition is not a valid partition of the nodes of the graph.
+
+    References
+    ----------
+    .. [1] Kernighan, B. W.; Lin, Shen (1970).
+       "An efficient heuristic procedure for partitioning graphs."
+       *Bell Systems Technical Journal* 49: 291--307.
+       Oxford University Press 2011.
+    .. [2] M. E. J. Newman,
+       "Modularity and community structure in networks",
+       PNAS, 103 (23), p. 8577-8582,
+       https://doi.org/10.1073/pnas.0601602103
+
+    """
+    n = len(G)
+    labels = list(G)
+    seed.shuffle(labels)
+    index = {v: i for i, v in enumerate(labels)}
+
+    if partition is None:
+        side = [0] * (n // 2) + [1] * ((n + 1) // 2)
+    else:
+        try:
+            A, B = partition
+        except (TypeError, ValueError) as err:
+            raise nx.NetworkXError("partition must be two sets") from err
+        if not nx.community.is_partition(G, [A, B]):
+            raise nx.NetworkXError("partition invalid")
+        side = [0] * n
+        for a in A:
+            side[index[a]] = 1
+
+    # ruff: noqa: E731   skips check for no lambda functions
+    # Using shortest_paths _weight_function with sum instead of min on multiedges
+    if callable(weight):
+        sum_weight = weight
+    elif G.is_multigraph():
+        sum_weight = lambda u, v, d: sum(dd.get(weight, 1) for dd in d.values())
+    else:
+        sum_weight = lambda u, v, d: d.get(weight, 1)
+
+    edges = [
```

Comment:
Let's change this name from `edges` to something like `list_of_nbr_eweight_tuples_by_node` or even `edges_info_by_node`